src/app/core/user.service.ts

```diff
@@ -26,15 +29,30 @@ export class UserService {
     });
   }
 
-  updateCurrentUser(value){
+  fetchUserData() {
+    return new Promise<any>( (resolve, reject) => {
+      this.database.list<any>(`investors/${firebase.auth().currentUser.uid}`)
+        .valueChanges()
+        .subscribe( submissions => {
```

Comment:
why observable in promise? just return only observable, without subscribe. Subscribe should be in component class. If you want to sort it, just map it for example:

...valueChanges().pipe(
 map(submissions => {
   return submissions.sort((n1, n2) => {
            if (n1.createdAt > n2.createdAt) { return 1; }
            if (n1.createdAt < n2.createdAt) { return -1; }
            return 0;
          });
}

  )
);